backend/replicache-transaction.ts

```diff
@@ -54,15 +63,38 @@ export class ReplicacheTransaction implements WriteTransaction {
     return val !== undefined;
   }
 
-  // TODO!
   async isEmpty(): Promise<boolean> {
-    throw new Error("not implemented");
-  }
-  scan(): ScanResult<string> {
-    throw new Error("not implemented");
+    for await (const _ of this.scan()) {
+      return false;
+    }
+    return true;
   }
-  scanAll(): Promise<[string, JSONValue][]> {
-    throw new Error("not implemented");
+
+  scan(options: ScanOptions = {} as ScanNoIndexOptions) {
+    if (isScanIndexOptions(options)) {
+      throw new Error("not implemented");
+    }
+
+    const { _executor: executor, _spaceID: spaceID, _cache: cache } = this;
+
+    return makeScanResult(options, async function* (fromKey) {
+      const source = getEntries(executor, spaceID, fromKey);
+
+      // TODO: It would be more optimal to keep the _cache in a sorted map in
+      // the first place.
+      const changes = [...cache]
+        .filter(([, { dirty }]) => dirty)
+        .filter(([k]) => k.localeCompare(fromKey) >= 0)
```

Comment:
`localeCompare` is not correct. It does crazy things that we do not want. We want non locale aware string compare

Reply:
What is the right JS function? Or do we need to use a library?

Reply:
I think what we really want is bytewise compare. And I guess that means to be correct the database has to use the same string encoding that JS does.